### PR TITLE
Ensure the behavior of `Map` is deterministic

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo-vm/Collections/OrderedDictionary.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+{
+    internal class OrderedDictionary<TKey, TValue> : IDictionary<TKey, TValue>
+    {
+        private class TItem
+        {
+            public TKey Key;
+            public TValue Value;
+        }
+
+        private class InternalCollection : KeyedCollection<TKey, TItem>
```
Comment (shargon) [inline src/neo-vm/Collections/OrderedDictionary.cs]: This is the faster one?

Comment (erikzhang) [reply src/neo-vm/Collections/OrderedDictionary.cs]: I just copied the code from https://github.com/neo-project/neo/blob/master/src/neo/IO/Caching/OrderedDictionary.cs

It is deterministic. We need it in VM.

Comment (shargon) [reply src/neo-vm/Collections/OrderedDictionary.cs]: I prefer to use SortedDictionary, but we can do benchmark later

Comment (erikzhang) [reply src/neo-vm/Collections/OrderedDictionary.cs]: `SortedDictionary` will sort the keys, which is not needed for us, and there will be additional overhead. We just need to arrange the items in the order they were added.

**Pull request merged**